File: code-coaches/hard/its-a-sign/its_a_sign.cpp
#include <iostream>
#include <vector>
#include <string>

bool is_palindrome(const std::string& word)
{
    int length = word.length();
    for (int i = 0; i < length / 2; i++)
    {
        if (word[i] != word[length - i - 1])
        {
            return false;
        }
    }
    return true;
}

bool contains_palindrome(const std::vector<std::string>& words)
{
    for (const auto& word : words)
    {
        if (is_palindrome(word))
        {
            return true;
        }
    }

    return false;
}

int main()
{
    const int size = 4;
    std::vector<std::string> words(size);

    for (int i = 0; i < size; i++)
    {
        std::cin >> words[i];
    }

    if (contains_palindrome(words)
        std::cout << "Open" << std::endl;
    else
        std::cout << "Trash" << std::endl;

    return 0;
}

File: code-coaches/hard/its-a-sign/ItsASign.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class ItsASign {

    public static boolean containsPalindrome(List<String> words) {
        for (String word : words) {
            if (isPalindrome(word)) {
                return true;
            }
        }
        return false;
    }

    public static boolean isPalindrome(String word) {
        StringBuilder reversed = new StringBuilder(word).reverse();
        return word.equals(reversed.toString());
    }

    public static void main(String[] args) {
        List<String> words = new ArrayList<>();

        Scanner scanner = new Scanner(System.in);
        for (int i = 0; i < 4; i++) {
            System.out.print("Enter word: ");
            String word = scanner.nextLine();
            words.add(word);
        }

        if (containsPalindrome(words)) {
            System.out.println("Open");
        } else {
            System.out.println("Trash");
        }

        scanner.close();
    }

}
